src/kratos.ts

```diff
@@ -160,7 +160,13 @@ export const dispatchWhoami = async (
 export const logout = async (
   req: FastifyRequest,
   res: FastifyReply,
+  isDeletion = false,
 ): Promise<FastifyReply> => {
+  const query = req.query as { reason?: string };
+  const reason = isDeletion
+    ? 'deletion logout'
+    : query?.reason || 'manual logout';
```

Comment:
We should validate `reason`, enum could help as mentioned on frontend.

Reply:
```ts
export enum LogoutReason {
  ManualLogout = 'manual logout',
  UserDeleted = 'user deleted',
  KratosSessionAlreadyAvailable = 'kratos session already available',
}

const reason =
    LogoutReason[
      isDeletion
        ? LogoutReason.UserDeleted
        : query?.reason || LogoutReason.ManualLogout
    ];
```

Did it like that. Haven't tested it yet, so not 100% if it actually works